### ISSUE TITLE
[@angular-eslint/template/interactive-supports-focus] Forms are erroneously marked as unfocusable

### ISSUE THREAD

**Issue by yktoo** [angular-eslint/angular-eslint#1429]
**Description and reproduction of the issue**

The rule erroneously marks the following form element as "Elements with interaction handlers must be focusable."

```HTML
<form (ngSubmit)="submit()" (keydown.control.enter)="submit()">
...
```

This is wrong because the handler is perfectly useful for submitting the form when **focus is in any of its controls**. The form itself is not required to have a tabindex, and it isn't meant to ever be focused itself.

In my opinion, this rule should check not only the element containing the handler, but all child elements as well, and defer if any of those is focusable. Keyboard events are [bubbling](https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event) by default:

> Keyboard events are only generated by `<input>`, `<textarea>`, `<summary>` and anything with the `contentEditable` or `tabindex` attribute. If not caught, they [bubble](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#event_bubbling) up the [DOM tree](https://developer.mozilla.org/en-US/docs/Web/API/Document_object_model/Using_the_Document_Object_Model#what_is_a_dom_tree) until they reach [Document](https://developer.mozilla.org/en-US/docs/Web/API/Document).

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `16.0.3` |
| `@angular-eslint/template-parser`         | `16.0.3` |
| `@typescript-eslint/parser`               | `5.60.0` |
| `ESLint`                                  | `8.43.0` |
| `node`                                    | `18.16.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 16.1.0
Node: 18.16.0
Package Manager: npm 9.5.1
OS: linux x64

Angular: 16.1.2
... animations, common, compiler, core, forms, localize
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1601.1
@angular-devkit/build-angular   16.1.1
@angular-devkit/core            16.1.0
@angular-devkit/schematics      16.1.0
@angular/cli                    16.1.0
@angular/compiler-cli           16.1.0
@schematics/angular             16.1.0
rxjs                            7.8.1
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by kgish** [comment]
I am getting the same issue with the following:

```
<mat-toolbar>
    <div class="footer-left">
        <span>...</span>
    </div>
    <div class="footer-center">
        <span>...</span>
    </div>
    <div class="footer-right">
        <span class="ui-icon ui-info" (click)="onClickInfo()" title="About..."></span>  <!-- HERE -->
    </div>
</mat-toolbar>

```

**Comment by localpcguy** [comment]
In addition to the examples listed, this can also apply to any place where the event listeners are listening for the events to bubble up. The individual items that can be clicked should be focusable, but the element that is listening to the bubbled events does not need to be in order for the functionality to be accessible.

**Comment by Ketec** [comment]
Just end up with same errors with that rule - and it does not even go away if the div has tabindex=0 or  role="button"
I am forced to add aria-disabled="true"... but it is not disabled - it is a clickable div based element.

div based buttons and other intractable elements are extremely common in web and found across all UX libraries.

**Comment by pedroll** [comment]
Same for <img>

**Comment by sandikbarr** [comment]
@pedroll can I ask why you have click or keyboard events on an img? It’s not the sort of element where you’d be capturing bubbled events. Should you be wrapping the img in a button for accessibility?
